docs/pages/project/contributing/contributing-error.md

```diff
@@ -9,27 +9,127 @@ category: contributing
+  If forced, all codes are renumbered. This can be useful to tidy up in earlier implementations of meshkit error codes.
+- Setting an error code to a call expression like ErrNoneDatabase = errors.NewDefault(ErrNoneDatabaseCode, "No Database selected")
+  is not allowed. This tool emits a warning if a call expression is detected.
+- Using errors.NewDefault(...) is deprecated. This tool emits a warning if this is detected.
+- Use errors.New(...) from meshkit to create actual errors with all the details.
+  This is often done in a factory function. It is important that the error code variable is used here, not a literal.
+  Specify detailed descriptions, probable causes, and remedies. They need to be string literals, call expressions are ignored.
+  This tool extracts this information from the code and exports it.
+- By convention, error codes and the factory functions live in files called error.go. The tool checks all files, but updates only error.go files.
+- This tool will create a couple of files, one of them is designed to be used to generate the error reference on the meshery website.
+  The file errorutil_analyze_summary.json contains a summary of the analysis, notably lists of duplicates etc.
+- The tool requires a file called component_info.json. Its location can be customized, by default it is the root directory (-d flag). 
+  This file has the following content, with concrete values specific for each component:
+  {
+    "name": "meshkit",
+    "type": "library",
+    "next_error_code": 11010
+  }
+- The tool updates next_error_code.
+
+
+## Example
+
+In this example we are trying to create an Error for not able to marhsal a JSON
 
 ```code
 var (
-    ErrExampleCode = "111"
+    // Error code 
+    ErrMarshalCode= "replace_me"
 
-    // Static errors
+    //Static errors (for example)
     ErrExample = errors.New(ErrExampleCode, errors.Alert, []string{"<short-description>"}, []string{"<long-description>"}, []string{"<probable-cause>"}, []string{"<suggested remediation>"})
 )
 
 // Dynamic errors
-func ErrExample(err error) error {
-    return errors.New(ErrExampleCode, errors.Alert, []string{"<short-description>"}, []string{"<long-description>"}, []string{"<probable-cause>"}, []string{"<suggested remediation>"})
+//Error Name
+func ErrMarshal(err error, obj string) error {
+	return errors.New(ErrMarshalCode, errors.Alert, []string{"Unable to marshal the : ", obj}, []string{err.Error()}, []string{}, []string{})
 }
+
 ```
+
+### Replacing old Error Codes 
+
+ Old
+ ```Code 
+    bd, err := json.Marshal(providers)
+	if err != nil {
+		http.Error(w, "unable to marshal the providers", http.StatusInternalServerError)
+		return
+	}
+ ```
+New
+  ```Code 
+    bd, err := json.Marshal(providers)
+    if err != nil {
+            obj := "provider"
+            http.Error(w, ErrMarshal(err, obj).Error(), http.StatusInternalServerError)
+            return
+        }
+ ```
+
+ 
+## Replacing logrus 
```

Comment:
Some wrong indentation is prolly creating weird code blocks. https://60efe6596f7390cc570636d8--meshery-docs-preview.netlify.app/project/contributing-error#replacing-logrus